### DISCUSSION TITLE
brew install gpsd does not install ubxtool

### DISCUSSION THREAD

**Discussion by bryanc-nz** [Homebrew/discussions#4698] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 128df14e0afd0525564051a7665708c2973a3cab
Last commit: 4 days ago
Core tap JSON: 04 Aug 22:52 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.o2KvYyozDj/org.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.5-x86_64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

The gpsd package does not include ubxtool.
I was able to build gpsd from source and install manually which did include ubxtool
To do this I first had to execute" 'pip install pyserial'

**Comment by osalbahr** [comment]
When you built `gpsd` from source, how did you do it differently than how [gpsd.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/gpsd.rb) does it?

**Reply by bryanc-nz** [reply]
I see that gpsd.rb depends on python@3.11, but it would seem that pyserial has to be installed separately, using 'pip install pyserial'. If pyserial is not installed then the ubxtool build will fail.

I don't see any attempt in gpsd.rb to check if pyserial is available.

**Reply by osalbahr** [reply]
> I see that gpsd.rb depends on python@3.11, but it would seem that pyserial has to be installed separately, using 'pip install pyserial'. If pyserial is not installed then the ubxtool build will fail.

To make sure I understand, does the documentation of `gpsd` say that you have to `pip install pyserial` *before* building `gpsd` from source to have it include `ubxtool`?

> I don't see any attempt in gpsd.rb to check if pyserial is available.

If you know how that looks like, FYI you can [Open a Pull Request](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request).

**Comment by bryanc-nz** [comment]
Yes - gpsd lists pyserial as a pre-requisite - see the pre-requisites at https://gpsd.gitlab.io/gpsd/installation.html#_check_your_installation_prerequisites

Sorry - I don't speak ruby and have no idea how to make the required changes

**Reply by osalbahr** [reply]
Since `pyserial` is not a Homebrew formula, not sure what the needed changes would be either.

```console
$ brew info pyserial
Error: No available formula with the name "pyserial".
==> Searching for a previously deleted formula (in the last month)...
Error: No previously deleted formula found.
```

**Reply by bryanc-nz** [reply]
Thanks for taking a look. I'll continue with my manual installation of gpsd.
